apps/main-landing/src/app/creators/donate/content.tsx

```diff
@@ -164,36 +157,32 @@ export const Content = ({ className }: Properties) => {
     'amount',
   ]);
 
-  const onChangeChainId = useCallback(
-    (chainId: number) => {
-      formMethods.resetField('tokenAddress', {
-        defaultValue: getDefaultTokenForChainId(chainId).address,
-      });
-    },
-    [formMethods],
-  );
 
   const allowedTokens = useMemo(() => {
-    const tokensForThisChain = CHAIN_ID_TO_TOKENS[chainId] ?? [];
-    const tokens = tokensForThisChain.filter((chainToken) => {
-      return possibleTokens.find((token) => {
-        return token.symbol === chainToken.symbol;
-      });
-    });
-    if (tokens.length === 0) {
-      return CHAIN_ID_TO_TOKENS[chainId] ?? [];
+    const allTokens = Object.values(CHAIN_ID_TO_TOKENS).flat();
+    const uniqueTokens: ChainToken[] = []
+    for(const token of allTokens) {
+      const exists = uniqueTokens.find((uniqueToken) => {
+        return uniqueToken.symbol === token.symbol
+      })
+      if(exists) {
+        continue
+      }
+      uniqueTokens.push(token)
     }
 
-    return tokens;
-  }, [possibleTokens, chainId]);
+    return uniqueTokens;
+  }, [possibleTokens]);
```

Comment:
Do we need `possibleTokens` in dependency array

Reply:
removed